woocommerce/woocommerce-gateway-stripe
# Deprecate manual webhook key input and introduce automatic webhook configuration

Part of #3195 

## Changes proposed in this Pull Request:

This PR replaces the old Webhook key input for button that can be used to automatically configure Webhooks for you. 

| Not configured | Configured |
|--------|--------|
| <img width="571" alt="Screenshot 2024-06-18 at 1 01 04 PM" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/be43d4c2-1944-4d52-ad2e-03e4bc81e0ff"> | <img width="547" alt="Screenshot 2024-06-18 at 1 00 02 PM" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/60ee1341-b458-4ad2-b6e9-a6019fdc1d67"> | 

## Testing instructions

### Previously entered webhook secret

1. On `develop` go to your Stripe API settings and enter a webhook secret. 
2. Checkout this branch.
3. run `npm run build`
4. Refresh the Stripe plugin settings page and open the api keys modal.
5. You should see the "Reconfigure webhooks" button. Notice that no URL is shown as we don't have any information that webhook. We only have the secret.

<p align="center">
<img width="280" alt="Screenshot 2024-06-18 at 3 00 46 PM" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/b05ad52c-fb0b-4bb8-8ebf-70471aceb49d">
</p>

### No webhook secret

1. Run the following code snippet to delete your webhook secrets. 

```php
$settings = get_option( 'woocommerce_stripe_settings', [] );

unset($settings['webhook_secret'] );
unset($settings['test_webhook_secret'] );

update_option( 'woocommerce_stripe_settings', $settings );
```

2. Refresh the Stripe settings page and open the API keys modal. 
3. You should now see the option to "Configure webhooks" button. 

<p align="center">
<img width="280" alt="Screenshot 2024-06-18 at 3 05 27 PM" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/50de8703-e192-4aae-8ddd-326c639aeff5">
</p>

4. Click the **Configure webhooks** button.
5. Once the request is handled you should now see the **Enabled** status, the button would have changed to "**Recongifure webhooks**" and information about where the webhooks will be sent is shown.
6. Clicking **Recongifure webhooks** will delete the old webhook and create a new one. You can verify this by checking your `woocommerce_stripe_settings` comparing the webhook secret or ID with the one in your Stripe Dashboard. 

### Some other things you may want to confirm/test


1. Entering a new secret key will use that secret key to configure the webhooks rather than the one stored in settings.
2. Emptying out the secret key field will disable the configure button.
3. While saving API keys, the configure button is disabled. 

---


**Post merge**
